src/main/javascript/crypto/src/eas/EasTicketAttestation.ts

```diff
@@ -541,9 +569,9 @@ export class EasTicketAttestation extends AttestableObject implements Attestable
 			chainId: this.signedAttestation.domain.chainId
 		};
 
-		const offchain = new Offchain(config);
+		const offchain = new Offchain(config, this.signedAttestation.message.version ?? 0);
 
-		const hash = ethers.utils._TypedDataEncoder.hash(offchain.getDomainTypedData(), {Attest: ATTESTATION_TYPE}, this.signedAttestation.message);
+		const hash = ethers.utils._TypedDataEncoder.hash(offchain.getDomainTypedData(), {Attest: OFFCHAIN_ATTESTATION_TYPES[this.signedAttestation.message.version].types}, this.signedAttestation.message);
```

Comment:
@micwallace , to make code more beautiful we can read string "Attest" from OFFCHAIN_ATTESTATION_TYPES[this.signedAttestation.message.version].primaryType

although this value doesnt matter here... we can use any name here and it will work correct

Reply:
Actually it doesn't work, seems there is a bug there in the EAS SDK. Even though v0 is specified with primaryType=Attestation, the attestations are actually issued with primaryType=Attest just like v1.

Relevant SDK code:
```
exports.OFFCHAIN_ATTESTATION_TYPES = {
    0: {
        domainName: 'EAS Attestation',
        primaryType: 'Attestation',
        types: [
            { name: 'schema', type: 'bytes32' },
            { name: 'recipient', type: 'address' },
            { name: 'time', type: 'uint64' },
            { name: 'expirationTime', type: 'uint64' },
            { name: 'revocable', type: 'bool' },
            { name: 'refUID', type: 'bytes32' },
            { name: 'data', type: 'bytes' }
        ]
    },
    1: {
        domainName: 'EAS Attestation',
        primaryType: 'Attest',
        types: [
            { name: 'version', type: 'uint16' },
            { name: 'schema', type: 'bytes32' },
            { name: 'recipient', type: 'address' },
            { name: 'time', type: 'uint64' },
            { name: 'expirationTime', type: 'uint64' },
            { name: 'revocable', type: 'bool' },
            { name: 'refUID', type: 'bytes32' },
            { name: 'data', type: 'bytes' }
        ]
    }
};
```

Reply:
The issue seems to be here:
```
const signedRequest = await this.signTypedDataRequest(params, {
            domain: this.getDomainTypedData(),
            primaryType: this.type.primaryType,
            message: params,
            types: {
                Attest: this.type.types
            }
        }, signer);
```

But this bug is even in older versions of EAS SDK, so it's nothing to worry about. 
You can see it for newly issued attestations here using an older library version: https://create.openloyalty.com/